diagnostic_feedback/quiz.py

```diff
@@ -241,6 +242,35 @@ def student_view(self, context=None):
             }
         )
 
+    def student_view_data(self, context=None):
+        """
+        Returns a JSON representation of the Diagnostic Feedback Xblock, that
+        can be retrieved using Course Block API.
+        """
+        return {
+            'quiz_type': self.quiz_type,
+            'quiz_title': self.title,
+            'questions': self.questions,
+        }
+
+    @XBlock.handler
+    def student_view_user_state(self, data, suffix=''):
+        """
+        Returns a JSON representation of the student data for Diagnostic Feedback Xblock
+        """
+        response = {
+            'student_choices': self.student_choices,
+            'student_result': self.student_result,
```

Comment:
yeah keeping `self.student_result` for this first version, we can change it in future if required.